packages/jellyfish-api-core/src/category/spv.ts

```diff
@@ -129,6 +129,32 @@ export class Spv {
   async listHtlcOutputs (scriptAddress?: string): Promise<ListHtlcsOutputsResult[]> {
     return await this.client.call('spv_listhtlcoutputs', [scriptAddress], { amount: 'bignumber' })
   }
+
+  /**
+   * Create, sign and send anchor tx, using only spv api
+   *
+   * @param {CreateAnchorInput[]} createAnchorInputs Info from BTC chain
+   * @param {string} createAnchorInput.txid The transaction id of the bitcoin UTXO to spent
+   * @param {string} createAnchorInput.vout The output index to spend in UTXO for tx fee
+   * @param {number} createAnchorInput.amount Amount of output in satoshis (base unit)
+   * @param {string} createAnchorInput.privkey WIF private key of bitcoin for signing this output
+   * @param {string} rewardAddress User's P2PKH address (in DeFi chain) for reward
+   * @param {CreateAnchorOptions} [options]
+   * @param {boolean} [options.send=true] Send it to BTC network
+   * @param {BigNumber} [options.feerate=1000] Feerate (satoshis) per KB, default is 1000.
+   * @return {Promise<CreateAnchorResult>}
+   */
+  async createAnchor (
+    createAnchorInputs: CreateAnchorInput[],
+    rewardAddress: string,
+    options: CreateAnchorOptions = { send: true, feerate: 1000 }
```

Comment:
In that case, if you set only one key, the other will not default

Reply:
@Jouzo  hmm... i don't quite get this... :thinking:

Reply:
As both keys are optional and default to a value, one should be able to call it with `{ send: false }` and have it default to `{ send: false, feerate: 100 }`. In that case, if you call it with one of the two keys, the other one won't default.

Reply:
yes.. understand.. i guess that `ain` has some reasons to default the keys
https://github.com/DeFiCh/ain/blob/ff53dcee23db2ffe0da9b147a0a53956f4e7ee31/src/spv/spv_rpc.cpp#L88-L89

Reply:
Yes it defaults on `ain` side as well. 
I still think it is better to set it on jellyfish side not to have unexpected behaviour later on

Reply:
i think i finally got what you mean
```
createSomething (details) {
  return { ...default, ...details }
}
```